### PR TITLE
Add sync_id to cat shards API

### LINKED ISSUE #14705: Expose sync id marker in _cat/shards API
Is it possible to expose the sync id in the _cat/shards API?
This would make is much easier for operators to check ids, than using the indices stats API.

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds the ability to get the sync_id from the cat shards API.

Closes #14705

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java [hunk 1]
```diff
@@ -91,6 +94,8 @@ protected Table getTableWithHeader(final RestRequest request) {
                 .addCell("id", "default:false;desc:unique id of node where it lives")
                 .addCell("node", "default:true;alias:n;desc:name of node where it lives");
 
+        table.addCell("commit.sync_id", "alias:sync;default:false;desc:sync id");
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java]: I wonder if the field name can just be sync_id and alias sync_id?

**Comment by bleskes** [comment]
LGTM. Left a minor suggestion. I think this can go into 2.2 as well, right?

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java]: Sure. Done in bf2783bb62b3c0746ad380178d9a55e7b42e6fa8.

**Comment by jasontedor** [comment]
> I think this can go into 2.2 as well, right?

Yes.

I pushed a REST test in 34c25dff424af0f0e4320103837881a1a55c3175. Can you take a look @bleskes?

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml [hunk 2]
```diff
@@ -86,6 +87,35 @@
...
+
+  - do:
+      cat.shards:
+        index: sync_id_test
+        h: index,sync_id
+  - match:
+      $body: |
+               /^(sync_id_test\s+[A-Za-z0-9_\-]{20}\s+\n){5}$/
+
+  - do:
```
Comment (bleskes) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: I don't think we need to delee.

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml [hunk 3]
```diff
@@ -86,6 +87,35 @@
...
+
+  - is_false: _shards.failed
+
+  - do:
+      cat.shards:
+        index: sync_id_test
+        h: index,sync_id
+  - match:
+      $body: |
+               /^(sync_id_test\s+[A-Za-z0-9_\-]{20}\s+\n){5}$/
```
Comment (bleskes) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: do we want a test of what happens when there is no sync id?

**Comment by bleskes** [comment]
Thx @jasontedor . Made minor comments

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml (hunk 2 above)
Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: @bleskes Without it, the rest of the tests have to change (either to explicitly list the indices excluding the index `sync_id_test`, or incorporate the additional output from the existence of this index). I think the tests should be independent though?

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml (hunk 3 above)
Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: @bleskes That seems like a good idea. Currently the output will be null for such shards. Before I write the test, do you think that is the output that we should go with?

Comment (bleskes) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: goog question. What do we do in other places? just space?

Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: @bleskes Yeah. Usually we leave null and then it just gets padded with spaces.

Comment (bleskes) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: That works, no?

Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: > That works, no?

Updated the test in 6eea2ed363b331b8baed0e255a505c041481d2e7.

Comment (bleskes) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: I meant create an index, don't call synced flush and see we return an empty field

Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yaml]: ACK. Added in 084385749f29a6cc6cf29338493a0146d1b9505d.

**Comment by bleskes** [comment]
LGTM, again.

**Comment by rtoma** [comment]
Once done, can you please backport to 1.7.x?

**Comment by jasontedor** [comment]
Integrated into [master](https://github.com/elastic/elasticsearch/commit/0a5323f4f7e2a12a16f4dc95f5b85edd52299571) and [2.x](https://github.com/elastic/elasticsearch/commit/9f1f376d057056f74cff44c802a4352b5871cad2) for inclusion in 2.2.0.

**Comment by jasontedor** [comment]
> Once done, can you please backport to 1.7.x?

@rtoma At this time, we are only pushing bug fixes to the 1.x line.

**Pull request merged**